### DISCUSSION TITLE
Coding Diary: Tic-Tac-Toe, Pandas & Sorting Algorithms

### DISCUSSION THREAD

**Discussion by Roadraudi** [Roadraudi/Codespace_Repository#13] [Announcements]
Willkommen zu meinem Coding-Blog! Heute teile ich meine neusten Erfahrungen in der Programmierung.

1. Tic-Tac-Toe: Ein fortlaufendes Projekt
Mein [Tic-Tac-Toe-Spiel](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Tic-Tac-Toe.py) nimmt Form an. In der letzten Session habe ich an Features und Fehlerkorrekturen gearbeitet. Es ist noch nicht fertig, aber in naher Zukunft werde ich dies auch fertig stellen!

2. Pandas: Schnelle Einblicke in die Datenanalyse
Um meine Datenanalysefähigkeiten zu verbessern, habe ich mich mit der Library Pandas beschäftigt und mal ein kleiner Datenframe erstellt. Bis jetzt noch nicht wirklich besonders aber dieses [Programm](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/pandaUebung.py) soll auch noch mit vielen weiteren coolen Features von Panda ausgestattet werden. Die folgende Seite hat mir persönlich weitergeholfen bei der Einführung in Pandas in Python: [Link](https://www.w3schools.com/python/pandas/pandas_dataframes.asp)

3. Sortieralgorithmen: Bubble Sort & Selection Sort
In meiner letzten Vorlesung über Theoretische Informatik sind wir näher auf Sortieralgorithmen eingegangen. Hier sind zwei davon:

[Bubble Sort](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/BubbleSort.py):
Ein einfacher, aber ineffizienter Sortieralgorithmus. Er vergleicht aufeinander folgende Elemente in der Liste und tauscht sie, wenn sie in der falschen Reihenfolge sind. Dieser Prozess wiederholt sich, bis die Liste sortiert ist.

[Selection Sort](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/SelectionSort.py):
Teilt die Liste in sortierte und unsortierte Teile. In jedem Durchlauf wird das kleinste Element aus dem unsortierten Teil ausgewählt und an die richtige Stelle im sortierten Teil verschoben.

Obwohl Python eine eingebaute sorted()-Funktion hat, die diese Algorithmen überflüssig macht, fand ich es dennoch interessant, die theoretischen Konzepte in praktischen Python-Code umzusetzen.

Das war mein Weekly-Code Update. 
Bis bald!

**Comment by salmuuki** [comment]
Nice, that you wanted to Share with us your new Skills of Algorithmen learned in Theoretische Informatik! Keep it going!

**Comment by NicoBaier123** [comment]
Sehr schön. Ich bin schon gespannt auf dein Panda Programm ;)
